xivo_bus/mixins.py

```diff
@@ -0,0 +1,421 @@
+# Copyright 2021 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import os
+import sys
+
+from threading import Thread, Lock, Semaphore
+from inspect import isclass
+from six import raise_from, iteritems
+from contextlib import contextmanager
+from collections import namedtuple, defaultdict
+from kombu import Queue, Exchange, Connection, Producer, binding as Binding
+from kombu.mixins import ConsumerMixin as KombuConsumer
+
+from .publishing_queue import PublishingQueue
+from .middlewares import Middleware, MiddlewareError
+
+
+class InitMixin(object):
+    def __init__(
+        self,
+        username='guest',
+        password='guest',
+        host='localhost',
+        port=5672,
+        exchange_name=None,
+        exchange_type=None,
+        middlewares=None,
+        **kwargs
+    ):
+        super(InitMixin, self).__init__(
+            username=username,
+            password=password,
+            host=host,
+            port=port,
+            exchange_name=exchange_name,
+            exchange_type=exchange_type,
+            middlewares=middlewares,
+            **kwargs
+        )
+
+
+class ThreadableMixin(object):
+    def __init__(self, **kwargs):
+        super(ThreadableMixin, self).__init__(**kwargs)
+        self.should_stop = False
```

Comment:
Did you check this implementation: https://github.com/wazo-platform/xivo-lib-python/blob/master/xivo/token_renewer.py#L24
Maybe dig in commits to find pro/con of this solution
But I doubt we use Event just for fun

Reply:
The current implementation was following the Kombu example, but the Event from threading library is indeed a much more robust solution

Reply:
The event wouldn't be very useful in this case because there is not waiting happening on it.  Token_renewer is using the event as it's timer to renew tokens